### PR TITLE
Handle ONNX model in Intel® OpenVINO quantization passes

### PR DESCRIPTION
**Pull request by Anirudh-Swaminathan**
Add ONNX model capabilities for Intel® OpenVINOQuantization and OpenVINOQuantizationWithAccuracy passes

Updated code to handle ONNXModelHandler for inputs and outputs for the Intel® OpenVINO quantization passes.
Updated unit tests for these passes as well.
Updated documentation for these passes.

- [x] Add unit tests for this change.
- [x] Make sure all tests can pass.
- [x] Update documents if necessary.
- [x] Lint and apply fixes to your code by running `lintrunner -a`
- [x] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [x] Is this PR including examples changes? If yes, please remember to update [example
documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

---------

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: olive/olive_config.json [hunk 1]
```diff
@@ -392,6 +392,15 @@
             "supported_quantization_encodings": [  ],
             "extra_dependencies": [ "openvino" ]
         },
+        "OpenVINOQuantizationWithAccuracy": {
+            "module_path": "olive.passes.openvino.quantization.OpenVINOQuantizationWithAccuracy",
+            "supported_providers": [ "*" ],
```
Comment (xiaoyu-work) [inline olive/olive_config.json:407]: Can the output model be run on all providers and devices?

File: test/requirements-test.txt [hunk 2]
```diff
@@ -25,6 +25,7 @@ onnx-graphsurgeon
 onnxconverter_common
 onnxmltools
 onnxoptimizer
+onnxruntime
```
Comment (xiaoyu-work) [inline test/requirements-test.txt]: Can you remove this? onnxruntime will be independently installed in our CI template.

File: olive/olive_config.json (hunk 1 above)
Comment (Anirudh-Swaminathan) [reply olive/olive_config.json:407]: Hey, this pass is actually not a new pass. It was never listed in olive_config.json previously.
OpenVINOQuantizationWithAccuracy is also just the exact same as the OpenVINOQuantization pass with extra accuracy controls by adding the ability to utilize a validation function to compute the validation set accuracy during the quantization process.
To directly address your question, both OpenVINOQuantization and OpenVINOQuantizationWithAccuracy passes are capable of outputting OpenVINO models as well as ONNX models, and ONNX models can theoretically be run on all providers and devices.

**Review by xiaoyu-work — DISMISSED** [review]

File: test/unit_test/passes/openvino/test_openvino_quantization.py [hunk 3]
```diff
@@ -57,6 +58,48 @@ def test_openvino_quantization(tmp_path):
     shutil.rmtree(data_dir)
 
 
+def test_openvino_quantization_onnx_input(tmp_path):
```
Comment (jambayk) [inline test/unit_test/passes/openvino/test_openvino_quantization.py:72]: the onnx tests are taking a long time to run (more than 20 minutes for each pass run). Please reduce the number of calibration data samples to speed up the process or use a small dummy dataset instead

**Review by jambayk — CHANGES_REQUESTED** [review]

File: test/unit_test/passes/openvino/test_openvino_quantization.py (hunk 3 above)
Comment (Anirudh-Swaminathan) [reply test/unit_test/passes/openvino/test_openvino_quantization.py:72]: Thanks for the feedback! I've reduced calibration dataset test set size in openvino quantization unit tests in the latest commit.

File: test/unit_test/passes/openvino/test_openvino_quantization.py [hunk 4]
```diff
@@ -12,16 +12,28 @@
...
     from torchvision.transforms import ToTensor
 
-    return CIFAR10(root=data_dir, train=False, transform=ToTensor(), download=True)
+    random.seed(1234)
+
+    # define the full CIFAR10 test set
+    full_test_set = CIFAR10(root=data_dir, train=False, transform=ToTensor(), download=True)
+
+    # randomply sample n_test_samples from the full test set
+    n_test_samples = 500
```
Comment (xiaoyu-work) [inline test/unit_test/passes/openvino/test_openvino_quantization.py]: The test still time out. It seems the test doesn't verify any output model. In that case, can we set this as 5 or 10 to just to test the functionality?

Comment (Anirudh-Swaminathan) [reply test/unit_test/passes/openvino/test_openvino_quantization.py]: OK, will reduce it to 5!
